File: daily-output.py
#!/usr/bin/env python3
#
# daily-output.py -- a sample script to fetch inverter output from SolarEdge API
# 
import requests
import json

# Fill in the two values below.
# See https://www.solaredge.com/sites/default/files/se_monitoring_api.pdf
SITE = ""
API_KEY = ""

# Request information from API.
response = requests.get("https://monitoringapi.solaredge.com/site/{0}/overview.json?api_key={1}".format(SITE, API_KEY))

# Show energy generated today along with date code. Run this after sundown to
# get the day's total output.
if (response.status_code == 200):
  print(response.text);
  data = json.loads(response.text);
  print(data['overview']['lastUpdateTime'], data['overview']['lastDayData']['energy'])
else:
  print("Error {0} connecting to API.".format(response.status_code))

File: daily-output.js
#!/usr/bin/env node

/*
 * solar-output.js -- a sample script to fetch inverter output from SolarEdge API
 */
 
const https = require("https");

// Fill in the two values below.
// See https://www.solaredge.com/sites/default/files/se_monitoring_api.pdf
const SITE = ""
const API_KEY = ""

// Make a call to the API.
let request = https.get(`https://monitoringapi.solaredge.com/site/${SITE}/overview.json?api_key=${API_KEY}`, (response) => {
  let buffer = [];

  // Piece together the response as it comes in.
  response.on('data', (chunk) => {
    buffer.push(chunk);
  });

  // When response is complete, show the results.
  response.on('end', (err) => {
    if (!err && response.statusCode == 200) {
      let data = JSON.parse(buffer);
      console.log(`${data.overview.lastUpdateTime} ${data.overview.lastDayData.energy}`);
    }
    else {
      console.error(`ERROR: ${err}`);
    }
  });

  // Complain if there was an error.
  response.on("error", (err) => {
    console.error(`ERROR: ${err}`);
  });
});
